consoleme/lib/oidc.py

```diff
@@ -54,24 +54,27 @@ async def populate_oidc_config():
         raise MissingConfigurationValue("Missing OIDC Secrets")
     oidc_config["client_id"] = client_id
     oidc_config["client_secret"] = client_secret
-    # Fetch jwks_uri for jwt validation
-    res = await http_client.fetch(
-        oidc_config["jwks_uri"],
-        method="GET",
-        headers={
-            "Content-Type": "application/x-www-form-urlencoded",
-            "Accept": "application/json",
-        },
-    )
-    oidc_config["jwks_data"] = json.loads(res.body)
     oidc_config["jwt_keys"] = {}
-    for k in oidc_config["jwks_data"]["keys"]:
-        key_type = k["kty"]
-        key_id = k["kid"]
-        if key_type == "RSA":
-            oidc_config["jwt_keys"][key_id] = RSAAlgorithm.from_jwk(json.dumps(k))
-        elif key_type == "EC":
-            oidc_config["jwt_keys"][key_id] = ECAlgorithm.from_jwk(json.dumps(k))
+    jwks_uris = [oidc_config["jwks_uri"]]
+    jwks_uris.extend(config.get("get_user_by_oidc_settings.extra_jwks_uri", []))
+    for jwks_uri in jwks_uris:
+        # Fetch jwks_uri for jwt validation
+        res = await http_client.fetch(
+            jwks_uri,
+            method="GET",
+            headers={
+                "Content-Type": "application/x-www-form-urlencoded",
+                "Accept": "application/json",
+            },
+        )
+        jwks_data = json.loads(res.body)
+        for k in jwks_data["keys"]:
+            key_type = k["kty"]
+            key_id = k["kid"]
+            if key_type == "RSA":
```

Comment:
Do we need to protect against `key_type = "none"` here? This isn't in the authn flow, but that might help prevent insecure configurations. Alternatively, we could slap on an `else` that raises an exception.

Reply:
Good call, added an exception here on unknown key type.